issues found by `nilness` static analysis tool

```
~/go/src/github.com/authzed/spicedb $ nilness ./...
/Users/dgryski/go/src/github.com/authzed/spicedb/internal/datastore/crdb/migrations/driver.go:88:9: impossible condition: nil != nil
/Users/dgryski/go/src/github.com/authzed/spicedb/pkg/development/devcontext.go:62:10: impossible condition: nil != nil
```

Both of these look like real issues:
From devcontext.go:
```
        if nerr != nil || devErrs != nil {
                // If any form of error occurred, immediately close the datastore
                derr := ds.Close()
                if err != nil {
                        return nil, nil, derr
                }

                return dctx, devErrs, nerr
```

From driver.go:
```
        result, err := tx.Exec(ctx, queryWriteVersion, version, replaced)
        if err != nil {
                return fmt.Errorf("unable to update version row: %w", err)
        }

        updatedCount := result.RowsAffected()
        if err != nil {
                return fmt.Errorf("unable to compute number of rows affected: %w", err)
        }
```

--- a/internal/datastore/crdb/migrations/driver.go
+++ b/internal/datastore/crdb/migrations/driver.go
@@ -85,10 +85,6 @@ func (apd *CRDBDriver) WriteVersion(ctx context.Context, tx pgx.Tx, version, rep
 	}
 
 	updatedCount := result.RowsAffected()
-	if err != nil {
-		return fmt.Errorf("unable to compute number of rows affected: %w", err)
-	}
-
 	if updatedCount != 1 {
 		return fmt.Errorf("writing version update affected %d rows, should be 1", updatedCount)
 	}

--- a/internal/datastore/crdb/reader.go
+++ b/internal/datastore/crdb/reader.go
@@ -213,7 +213,7 @@ func loadAllNamespaces(ctx context.Context, tx pgx.Tx) ([]*core.NamespaceDefinit
 	}
 
 	if rows.Err() != nil {
-		return nil, fmt.Errorf(errUnableToReadConfig, err)
+		return nil, fmt.Errorf(errUnableToReadConfig, rows.Err())
 	}
 
 	return nsDefs, nil

--- a/internal/datastore/mysql/migrations/driver.go
+++ b/internal/datastore/mysql/migrations/driver.go
@@ -85,7 +85,7 @@ func (driver *MySQLDriver) Version(ctx context.Context) (string, error) {
 	}
 	defer LogOnError(ctx, rows.Close)
 	if rows.Err() != nil {
-		return "", fmt.Errorf("unable to load driver migration revision: %w", err)
+		return "", fmt.Errorf("unable to load driver migration revision: %w", rows.Err())
 	}
 	cols, err := rows.Columns()
 	if err != nil {

--- a/internal/graph/lookupsubjects.go
+++ b/internal/graph/lookupsubjects.go
@@ -100,7 +100,7 @@ func (cl *ConcurrentLookupSubjects) lookupDirectSubjects(
 	var foundSubjectIds []string
 	for tpl := it.Next(); tpl != nil; tpl = it.Next() {
 		if it.Err() != nil {
-			return err
+			return it.Err()
 		}
 
 		if tpl.Subject.Namespace == req.SubjectRelation.Namespace &&
@@ -187,7 +187,7 @@ func (cl *ConcurrentLookupSubjects) lookupViaTupleToUserset(
 	toDispatchByTuplesetType := tuple.NewONRByTypeSet()
 	for tpl := it.Next(); tpl != nil; tpl = it.Next() {
 		if it.Err() != nil {
-			return err
+			return it.Err()
 		}
 
 		toDispatchByTuplesetType.Add(tpl.Subject)

--- a/internal/services/v1/relationships.go
+++ b/internal/services/v1/relationships.go
@@ -132,7 +132,7 @@ func (ps *permissionServer) ReadRelationships(req *v1.ReadRelationshipsRequest, 
 		}
 	}
 	if tupleIterator.Err() != nil {
-		return status.Errorf(codes.Internal, "error when reading tuples: %s", err)
+		return status.Errorf(codes.Internal, "error when reading tuples: %s", tupleIterator.Err())
 	}
 
 	return nil

--- a/pkg/development/devcontext.go
+++ b/pkg/development/devcontext.go
@@ -59,7 +59,7 @@ func NewDevContext(ctx context.Context, requestContext *devinterface.RequestCont
 	if nerr != nil || devErrs != nil {
 		// If any form of error occurred, immediately close the datastore
 		derr := ds.Close()
-		if err != nil {
+		if derr != nil {
 			return nil, nil, derr
 		}
 

